src/libfetchers/github.cc

```diff
@@ -172,20 +172,53 @@ struct GitArchiveInputScheme : InputScheme
         return input;
     }
 
-    std::optional<std::string> getAccessToken(const fetchers::Settings & settings, const std::string & host) const
+    // Search for the longest possible match starting from the begining and ending at either the end or a path segment.
+    std::optional<std::string> getAccessToken(const fetchers::Settings & settings, const std::string & host, const std::string & url) const override
     {
         auto tokens = settings.accessTokens.get();
+        std::string answer;
+        size_t answer_match_len = 0;
+        if(! url.empty()) {
+            for (auto & token : tokens) {
+                auto first = url.find(token.first);
+                if (
```

Comment:
Does this still work? If `url` is e.g. `https://github.com/...`, then it won't match with a token for `github.com` anymore, right? You may need to use `parseURL()` to get rid of the scheme part of the URL.

Reply:
Perhaps that is a misnomer. The "url" is made here: https://github.com/NixOS/nix/pull/12465/files#diff-be2f63dc22d73bd86ee70820979ee54ff2bb5cef76e3bbb53a673fddfd2a1b5cR211